WordPress/gutenberg
# Add the ability to create a new page in the Site Editor

## What?
As part of https://github.com/WordPress/gutenberg/issues/44461 we are re-introducing a way to edit content within the site editor. Adding new content, such as pages, is an important part of that experience. This PR gives us a very basic starting experience to work from. 

## Why?
Resolves first iteration of https://github.com/WordPress/gutenberg/issues/48456

## How?
The AddNewPageModal sits above site view / editor and adds open state to store so that it can be called upon in different places (e.g. navigation block). In future this might be more appropriate as a package.

TODO: We'll want the command center "new page" command to make use of this modal as well.

## Testing Instructions
1. Open site editor
2. Click on pages
3. Click the `+` button
4. Notice new page modal opens
5. Give page a title and hit create
6. You should be redirected to editor with new page loaded

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1072756/596edf2e-9756-456b-8b26-b2dabea28828
